doc/specs/Keyboard-Selection.md

```diff
@@ -0,0 +1,141 @@
+
+For **vertical movements**, the movement is a little more complicated than before. The selection will still respond to buffer and viewport boundaries as before. If the user is trying to move up, the selection anchor will attempt to move up by one line, then selection will be expanded leftwards. Alternatively, if the user is trying to move down, the selection anchor will attempt to move down by one line, then the selection will be expanded rightwards.
+
+For `SelectionExpansionMode = Viewport`, the selection anchor will be updated according to the viewport's height. Horizontal movements will be updated according to the viewport's width, thus resulting in the anchor being moved to the left/right boundary of the viewport.
+
+**NOTE**: An important thing to handle properly in all cases is wide glyphs. The user should not be allowed to select a portion of a wide glyph; it should be all or none of it. When calling `_ExpandWideGlyphSelection` functions, the result must be saved to the anchor.
+
+**NOTE**: In all cases, horizontal movements attempting to move past the left/right viewport boundaries result in a wrap. Vertical movements attempting to move past the top/bottom viewport boundaries will scroll such that the selection is at the edge of the screen. Vertical movements attempting to move past the top/bottom buffer boundaries will be clamped to be within buffer boundaries.
+
+Every combination of the `Direction` and `SelectionExpansionMode` will map to a keybinding. These pairings are shown below in the UI/UX Design --> Keybindings section.
+
+### Mark Mode
+Upon mark mode being enabled by a user keybinding, the current cursor position becomes the selection anchor. The arrow keys move the selection anchor by cell.
+
+
+## UI/UX Design
+
+### Keybindings
+
+By default, the following keybindings will be set:
+| Action | Keychord | `KeyboardSelection(Direction, SelectionExpansionMode)` Parameters |
+|--|--|--|
+| `MoveSelectionAnchorUp`              | `Shift` + `↑`              | `Up`      , `Cell`
+| `MoveSelectionAnchorDown`            | `Shift` + `↓`              | `Down`    , `Cell`
+| `MoveSelectionAnchorLeft`            | `Shift` + `←`              | `Left`    , `Cell`
+| `MoveSelectionAnchorRight`           | `Shift` + `→`              | `Right`   , `Cell`
+| `MoveSelectionAnchorUpOneScreen`     | `Shift` + `Page Up`        | `Up`      , `Viewport`
+| `MoveSelectionAnchorDownOneScreen`   | `Shift` + `Page Down`      | `Down`    , `Viewport`
+| `MoveSelectionAnchorToLeftMargin`    | `Shift` + `Home`           | `Left`    , `Viewport`
+| `MoveSelectionAnchorToRightMargin`   | `Shift` + `End`            | `Right`   , `Viewport`

```

Comment:
mild worry about these default `shift+key` keybindings. Does vim, emacs bind any of them by default?

Reply:
It's common for tmux users to bind tab navigation to shift+arrow keys.

Reply:
Oof. The good news is that these keybindings would only get caught when a selection is active. But would that be good enough?

Reply:
@carlos-zamora That should be fine as long as you don't step on tmux's own selection mode, since selection mode would never be activated because tmux has its own built-in implementation.

tmux also emits its internal clipboard using special VT sequences, so it would be nice if Terminal supported that (your built-in selection mode will not work with tmux or screen for very long selections, since they have their own built-in buffers).